cmd/unlockerd/main.go

```diff
@@ -0,0 +1,260 @@
+	if len(parts) != 2 {
+		return fmt.Errorf("%s must be a valid address in the form host:port", rpcAddressKey)
+	}
+	if parts[0] != "" && parts[0] != "localhost" {
+		if net.ParseIP(parts[0]) == nil {
+			return fmt.Errorf("%s host is invalid", rpcAddressKey)
+		}
+	}
+	port, err := strconv.Atoi(parts[1])
+	if err != nil || port <= 1024 {
+		return fmt.Errorf("%s port is invalid", rpcAddressKey)
+	}
+
+	provider := viper.GetString(providerKey)
+	if provider == "" {
+		return fmt.Errorf("%s must not be null", providerKey)
+	}
+	if _, ok := supportedProviders[provider]; !ok {
+		return fmt.Errorf(
+			"unknown provider '%s', must be one of [%s]",
+			provider, strings.Join(supportedProviders.keys(), ","),
+		)
+	}
+
+	interval := viper.GetInt(intervalKey)
+	if interval <= 0 {
+		return fmt.Errorf("%s must be a positive number", intervalKey)
+	}
+
+	return nil
+}
+
+func main() {
+	rpcAddress := viper.GetString(rpcAddressKey)
+	providerType := viper.GetString(providerKey)
+	interval := time.Duration(viper.GetInt(intervalKey)) * time.Second
+
+	prov, err := supportedProviders[providerType]()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	password, err := prov.Password()
+	if err != nil {
+		log.Fatalf("error while sourcing password: %s", err)
+	}
+	tlsCertificate, err := prov.TLSCertificate()
+	if err != nil {
+		log.Fatalf("error while sourcing TLS certificate: %s", err)
+	}
+
+	client, cleanup, err := getUnlockerClient(rpcAddress, tlsCertificate)
+	if err != nil {
+		log.Fatalf("error while setting up gRPC client: %s", err)
+	}
+	defer cleanup()
+
+	status, err := getWalletStatus(client)
+	if err != nil {
+		log.Fatalf("error while retrieving wallet status: %s", err)
+	}
+	if !status.Initialized {
+		log.Info("waiting for the wallet to be initialized")
+	}
+	for !status.Initialized {
+		time.Sleep(interval)
+
+		status, err := getWalletStatus(client)
+		if err != nil {
+			log.Fatalf("error while retrieving wallet status: %s", err)
+		}
+		if status.Initialized {
+			break
+		}
+	}
+	log.Info("wallet initialized")
+
+	if status.Unlocked {
+		log.Info("wallet is already unlocked. Nothing left to do")
+		return
+	}
+
+	log.Info("Attempting to unlock it with provided password...")
+
+	if _, err := client.UnlockWallet(context.Background(), &pbunlocker.UnlockWalletRequest{
+		WalletPassword: password,
+	}); err != nil {
+		log.Fatalf("error while unlocking wallet: %s", err)
+	}
+
+	log.Info("wallet unlocked successfully!")
+}
+
+func getWalletStatus(client pbunlocker.WalletUnlockerClient) (*pbunlocker.IsReadyReply, error) {
+	ctx := context.Background()
+	return client.IsReady(ctx, &pbunlocker.IsReadyRequest{})
+}
+
+func pathExists(path string) bool {
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
+
+func cleanAndExpandPath(path string) string {
+	if path == "" {
+		return ""
+	}
+
+	// Expand initial ~ to OS specific home directory.
+	if strings.HasPrefix(path, "~") {
```

Comment:
why we do not use btcutil HomeDir method? which is the same used by the daemon to create his datadir?

Reply:
I actually copied the `cleanAndExpandPath` method from the [cli](../blob/master/cmd/tdex/main.go#L258).